adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1780,6 +1792,65 @@ private void checkInternetPermission() {
     }
 
     /**
+     * To verify if App gives permissions to AccountManager to use broker.
+     *
+     * @throws UsageAuthenticationException
+     * @return boolean If the GET_ACCOUNTS, MANAGE_ACCOUNTS, USE_CREDENTIALS permissions are granted in the Manifest.xml
+     */
+    private boolean verifyManifestPermissions() throws UsageAuthenticationException {
+        if(Build.VERSION.SDK_INT <= 22) {
+            final String methodName = ":verifyManifestPermissions";
+            final PackageManager packageManager = mContext.getPackageManager();
+            final StringBuilder permissionMissing = new StringBuilder();
+            Logger.v(
+                    TAG + methodName,
+                    "Broker related permissions checking starts.");
+
+            if(PackageManager.PERMISSION_GRANTED != packageManager.checkPermission(
+                    "android.permission.GET_ACCOUNTS", mContext.getPackageName())) {
+                permissionMissing.append("GET_ACCOUNTS ");
+                Logger.w(
+                        TAG + methodName,
+                        "Broker related permissions are missing for GET_ACCOUNTS",
+                        "", ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING);
+            }
+
+            if(PackageManager.PERMISSION_GRANTED != packageManager.checkPermission(
+                    "android.permission.MANAGE_ACCOUNTS", mContext.getPackageName())) {
+                permissionMissing.append("MANAGE_ACCOUNTS ");
+                Logger.w(
+                        TAG + methodName,
+                        "Broker related permissions are missing for MANAGE_ACCOUNTS",
+                        "", ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING);
+            }
+
+            if(PackageManager.PERMISSION_GRANTED != packageManager.checkPermission(
+                    "android.permission.USE_CREDENTIALS", mContext.getPackageName())) {
+                permissionMissing.append("USE_CREDENTIALS");
+                Logger.w(
+                        TAG + methodName,
+                        "Broker related permissions are missing for USE_CREDENTIALS",
+                        "", ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING);
+            }
+
+            Logger.v(
+                    TAG + methodName,
+                    "Broker related permissions are verified");
+
+            if(permissionMissing.length() == 0) {
+                return true;
+            } else {
```

Comment:
there is no need to have a return here, just do a verify, if it fails, just throw.